packages/uhk-web/src/app/services/theme.service.ts

```diff
@@ -0,0 +1,98 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { fromEvent, Observable } from 'rxjs';
+import { skipWhile, withLatestFrom } from 'rxjs/operators';
+import { LogService, AppTheme } from 'uhk-common';
+import { AppState, getAppTheme } from '../store';
+
+const THEME_FILES = {
+    DEFAULT: 'stylesLight.css',
+    [AppTheme.Light]: 'stylesLight.css',
+    [AppTheme.Dark]: 'stylesDark.css'
+};
+
+const UHK_THEME_ID = 'uhk-theme';
+
+@Injectable({ providedIn: 'root' })
+export class ThemeService {
+    appTheme$: Observable<AppTheme>;
+    darkModeListener$: Observable<MediaQueryListEvent>;
+    darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+
+    constructor(
+        @Inject(DOCUMENT) private document: Document,
+        private store: Store<AppState>,
+        private logService: LogService
+    ) {
+        this.appTheme$ = this.store.select(getAppTheme);
+        this.appTheme$.subscribe(value => this.setTheme(value));
```

Comment:
Please implement the `OnDestroy` interface and unsubscribe. I know it is a singleton in the root DI, but I like if we properly release the resources, prevent us from the future memory leak investigations

Reply:
Done.